orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -173,21 +197,25 @@ def forward(ctx, *inputs, **kwargs):
 
                     # TODO: Try to reuse the output buffers as some of the output tensors are same sizes,
                     #   especially the backward graph outputs.
-                    training_io_binding = self._training_session.io_binding()
+                    execution_session = self._training_session if use_training_session else self._inference_session
+                    io_binding = execution_session.io_binding()
                     run_options = C.RunOptions()
-                    
+                    run_options.training_mode = training_mode
+
                     # Use IO binding
-                    _create_iobinding(training_io_binding, inputs, self._onnx_training, self._device)
+                    _create_iobinding(io_binding, inputs,
+                        self._onnx_training if use_training_session else self._onnx_inference,
+                        self._device, training_mode)
 
                     # Run and return module outputs.
-                    forward_outputs, run_id = self._training_session.run_forward(training_io_binding, run_options)
+                    forward_outputs, run_id = execution_session.run_forward(io_binding, run_options)
```

Comment:
I'm not sure if it matters, but I would have expected the inference-mode call to happen outside the `torch.autograd.Function`. (At a guess, perhaps this causes the returned tensors to have an associated gradient function, which might consume more memory than needed?)